File: Project_data/University/Algoritm/semertr_3/kursovaya/task_4.py
import numpy as np
from scipy import integrate 

# Определяем функции
def f1(x):
    return x / (x**2 + 6*x + 1)

def f2(x):
    return (10**(0.01*x)) * np.cos(0.125*x)

def f3(x):
    return np.arccos((3*x + 25) / 100)

# Функция для вычисления интеграла методом Симпсона
def compute_integral(func, a, b):
    result, error = integrate.quad(func, a, b)
    return result

lower_base = int(input('Введите нижнее основание интеграла: '))
upper_base = int(input('Введите верхнее основание интеграла: '))

# Используем функции
print("Интеграл f1 от 0 до 1: ", compute_integral(f1, lower_base, upper_base))
print("Интеграл f2 от 0 до 1: ", compute_integral(f2, lower_base, upper_base))
print("Интеграл f3 от 0 до 1: ", compute_integral(f3, lower_base, upper_base))

File: Project_data/University/Algoritm/semertr_3/kursovaya/task_4.c
#include <stdio.h>
#include <math.h>
#include <string.h>

// Определяем заданные функции
double f1(double x) {
    return x / (x * x + 6 * x + 1);
}

double f2(double x) {
    return pow(10, -0.01 * x) * cos(0.125 * x);
}

double f3(double x) {
    return acos((3 * x + 25) / 100);
}

// Функция для вычисления интеграла методом Симпсона
double compute_integral(double (*func)(double), double a, double b, int n) {
    double h = (b - a) / n;
    double sum = func(a) + func(b);
    
    for (int i = 1; i < n; i++) {
        double x = a + i * h;
        sum += (i % 2 == 0) ? 2 * func(x) : 4 * func(x);
    }
    
    return (h / 3) * sum;
}

int main() {
    double lower_base, upper_base;
    char num_func[100];
    
    // Ввод пределов интегрирования
    printf("Введите нижнее основание интеграла: ");
    scanf("%lf", &lower_base);
    printf("Введите верхнее основание интеграла: ");
    scanf("%lf", &upper_base);
    
    int n = 1000; // Количество разбиений для интегрирования
    
    // Выбор функции интегрирования
    printf("Выберите функцию, для которой будет расчитан интеграл (f1, f2, f3): ");
    scanf("%s", num_func);

    if (strcmp(num_func, "f1") == 0) {
        printf("Интеграл f1 от %.2f до %.2f: %.6f\n", lower_base, upper_base, compute_integral(f1, lower_base, upper_base, n));
    } else if (strcmp(num_func, "f2") == 0) {
        printf("Интеграл f2 от %.2f до %.2f: %.6f\n", lower_base, upper_base, compute_integral(f2, lower_base, upper_base, n));
    } else {
        printf("Интеграл f3 от %.2f до %.2f: %.6f\n", lower_base, upper_base, compute_integral(f3, lower_base, upper_base, n));
    }   
    
    return 0;
}